src/main/java/org/graylog/integrations/notifications/types/SlackEventNotification.java

```diff
@@ -35,148 +36,160 @@
+            SlackMessage slackMessage = createSlackMessage(ctx, config);
+            slackClient.send(slackMessage, config.webhookUrl());
+        } catch (Exception e) {
+
+            LOG.error("SlackEventNotification send error for id {} : {}", ctx.notificationId(), e.toString());
+            final Notification systemNotification = notificationService.buildNow()
+                    .addNode(nodeId.toString())
+                    .addType(Notification.Type.GENERIC)
+                    .addSeverity(Notification.Severity.NORMAL)
+                    .addDetail("SlackEventNotification send error ", e.toString());
+
+            notificationService.publishIfFirst(systemNotification);
+            throw new TemporaryEventNotificationException("Slack notification is triggered, but sending failed. " + e.getMessage(), e);
+
+        }
+
+    }
+
+    /**
+     * @param ctx
+     * @param config
+     * @return
+     * @throws PermanentEventNotificationException - throws this exception when the custom message template is invalid
+     */
+    SlackMessage createSlackMessage(EventNotificationContext ctx, SlackEventNotificationConfig config) throws PermanentEventNotificationException {
+        //Note: Link names if notify channel or else the channel tag will be plain text.
+        boolean linkNames = config.linkNames() || config.notifyChannel();
+        String message = buildDefaultMessage(ctx, config);
+
+        String customMessage = null;
+        String template = config.customMessage();
+        if (!isNullOrEmpty(template)) {
+            customMessage = buildCustomMessage(ctx, config, template);
+        }
+
+        return new SlackMessage(
+                config.color(),
+                config.iconEmoji(),
+                config.iconUrl(),
+                config.userName(),
+                config.channel(),
+                linkNames,
+                message,
+                customMessage
+        );
+    }
+
+    String buildDefaultMessage(EventNotificationContext ctx, SlackEventNotificationConfig config) {
+        String title = buildMessageTitle(ctx);
+
+        // Build custom message
+        String audience = config.notifyChannel() ? "@channel " : "";
+        String description = ctx.eventDefinition().map(EventDefinitionDto::description).orElse("");
+        return String.format(Locale.ROOT, "%s*Alert %s* triggered:\n> %s \n", audience, title, description);
+    }
+
+    private String buildMessageTitle(EventNotificationContext ctx) {
+        String eventDefinitionName = ctx.eventDefinition().map(EventDefinitionDto::title).orElse("Unnamed");
+        return "_" + eventDefinitionName + "_";
+    }
+
+    String buildCustomMessage(EventNotificationContext ctx, SlackEventNotificationConfig config, String template) throws PermanentEventNotificationException {
+        final List<MessageSummary> backlog = getAlarmBacklog(ctx);
+        final List<MessageSummary> truncatedBacklog = getMessageBacklog(config, backlog);
```

Comment:
Why have a separate method for truncating the backlog?  Wouldn't it be simpler to pass the config or a limit into the `getAlarmBacklog` method and just get the appropriate backlog returned?

Reply:
please refer to bernd's earlier comments, he prefers a `immutable` list

Reply:
And since you have control over what will be returned by `getAlarmBacklog`, that method can return an immutable list if desired.